src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalFilterLongReadAlignmentsSAMSpark.java

```diff
@@ -0,0 +1,328 @@
+                            return weight * len;
+                        })
+                        .sum();
+
+        int redundancy = 0;
+        for (int i = 0; i < configuration.size() -1 ; ++i) {
+            final int b1 = configuration.get(i).endInAssembledContig;
+            for (int j = i + 1; j < configuration.size(); ++j) {
+                final int a2 = configuration.get(j).startInAssembledContig;
+                final int overlap = Math.max(b1 - a2 + 1, 0);
+                redundancy += overlap;
+            }
+        }
+
+        return tigExplainQual - redundancy;
+    }
+
+    /**
+     * Depending on the number of best-scored configurations, one contig may produce multiple contigs with the same name
+     * and sequence but different selected configurations.
+     */
+    private static Iterator<AlignedContig> reConstructContigFromPickedConfiguration(
+            final Tuple2<String, Tuple2<byte[], List<List<AlignmentInterval>>>> forOneContig) {
+        if (forOneContig._2._2.size() > 1) {
+            return forOneContig._2._2.stream()
+                    .map(configuration -> new AlignedContig(forOneContig._1, forOneContig._2._1,
+                            splitGaps(configuration), true))
+                    .sorted(sortConfigurations())
+                    .collect(Collectors.toList()).iterator();
+        } else {
+            return Collections.singletonList(new AlignedContig(forOneContig._1, forOneContig._2._1,
+                    splitGaps(forOneContig._2._2.get(0)), false))
+                    .iterator();
+        }
+    }
+
+    /**
+     * when two configurations are the same, prefer the one with less alignments,
+     * or less summed mismatches if still tie.
+     */
+    private static Comparator<AlignedContig> sortConfigurations() {
+        Comparator<AlignedContig> numFirst
+                = (AlignedContig x, AlignedContig y) -> Integer.compare(x.alignmentIntervals.size(), y.alignmentIntervals.size());
+        Comparator<AlignedContig> mismatchSecond
+                = (AlignedContig x, AlignedContig y) -> Integer.compare(x.alignmentIntervals.stream().mapToInt(ai -> ai.mismatches).sum(),
+                                                                        y.alignmentIntervals.stream().mapToInt(ai -> ai.mismatches).sum());
+        return numFirst.thenComparing(mismatchSecond);
+    }
+
+    private static List<AlignmentInterval> splitGaps(final List<AlignmentInterval> configuration) {
+        return configuration.stream()
+                .map(ai -> GappedAlignmentSplitter.split(ai, GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY,
+                        SvCigarUtils.getUnclippedReadLength(ai.cigarAlong5to3DirectionOfContig)))
+                .flatMap(Utils::stream).collect(Collectors.toList());
+    }
+
+    /**
+     * Primary reference contigs are defined as chromosomes 1-22, X, Y, M, and defined for both GRCh38 and hg19.
+     */
+    static Set<String> getCanonicalChromosomes() {
```

Comment:
I've wonderer about what that input is....
Done as suggested, and added the necessary cmd line argument , though left option for such file to be missing (hence optional cmd line argument) and default to human.